src/docfx/build/toc/TableOfContentsLoader.cs

```diff
@@ -113,12 +151,12 @@ private void TraverseAndMerge(TableOfContentsNode node, List<SourceInfo<TableOfC
 
             foreach (var item in node.Items)
             {
-                TraverseAndMerge(item, itemsToMatch, matched);
+                TraverseAndMerge(item, itemsToMatch, matched, joinTOCConfig);
             }
         }
 
         private TableOfContentsNode LoadTocFile(
-            FilePath file, FilePath rootPath, List<FilePath> referencedFiles, List<FilePath> referencedTocs)
+            FilePath file, FilePath rootPath, List<FilePath> referencedFiles, List<FilePath> referencedTocs, bool loadTocNodes = false)
```

Comment:
consider remove the default value, actually the default behavior should be `true`